File: backend/app/Models/Product.php
<?php

namespace App\Models;

use App\Support\Auditable;
use App\Support\BelongsToCompany;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use Auditable, BelongsToCompany, HasFactory, SoftDeletes;

    protected $fillable = [
        'company_id', 'category_id', 'sku', 'barcode', 'name', 'description',
        'unit', 'is_active', 'price', 'cost', 'weight', 'volume',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'price' => 'decimal:2',
        'cost' => 'decimal:2',
        'weight' => 'decimal:3',
        'volume' => 'decimal:3',
    ];

    public function category()
    {
        return $this->belongsTo(ProductCategory::class, 'category_id');
    }
}

File: mobile/lib/features/products/domain/product.dart
class Product {
  const Product({required this.id, required this.sku, required this.name, required this.unit});

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'] as int,
        sku: json['sku'] as String,
        name: json['name'] as String,
        unit: json['unit'] as String,
      );

  final int id;
  final String sku;
  final String name;
  final String unit;

  @override
  String toString() => name;
}
